go/provisioner/local/local.go

```diff
@@ -763,6 +755,14 @@ func (p *localProvisioner) provisionMultigateway(ctx context.Context, req *provi
 		fmt.Printf("Warning: failed to save service state: %v\n", err)
 	}
 
+	// Wait for multigateway to be ready
```

Comment:
@deepthi  - This makes sure that this connectivity checks happen after the process state file is saved. It should fix the issue you found

This is tricky to add an integration test though. Do you have thoughts on that?

Reply:
Agree that it is tricky. For now it should be fine to simulate the failure locally and make sure it works.